### PR TITLE
chore: Bump GH macOS runner

### PR DESCRIPTION
**Pull request by bitsandfoxes**
On December 3rd, 2024, `macos-12` runners will be fully unsupported.
GH actions issue: https://github.com/actions/runner-images/issues/10721

#skip-changelog

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -602,21 +602,21 @@ jobs:
...
+        # - `iOS 17`: 86 %
+        # - `iOS 16`: 11 %
+        # - the rest:  3 %
+        # as of October, 2024. Therefore, let's stick to testing iOS 16 and `latest` for now.
         # Numbers as string otherwise GH will reformat the runtime numbers removing the fractions.
         # Also make sure to match the versions available here:
-        #  - https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md
-        ios: ["15.4", latest] # last updated May 2023
+        #  - https://github.com/actions/runner-images/blob/main/images/macos/macos-13-Readme.md
+        ios: ["16.1", latest] # last updated October 2024
```
Comment (philipphofmann) [inline .github/workflows/ci.yml:619]: `m`: Xcode latest on macos-13 is iOS `17.2` if I'm not mistaken: https://github.com/actions/runner-images/blob/main/images/macos/macos-13-Readme.md#installed-sdks

**Review by philipphofmann — APPROVED** [review]
When CI is green, LGTM.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (bitsandfoxes) [reply .github/workflows/ci.yml:619]: Am I looking at the wrong place? The [runner lists 16.1](https://arc.net/l/quote/ndshmaxl)?

**Review by philipphofmann — APPROVED** [review]
LGTM, but I'm not sure if CI is happy yet.

**Pull request merged**